x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import seedrandom from 'seedrandom';
+import { EndpointDocGenerator, TreeOptions, Event } from './generate_data';
+
+export async function indexHostsAndAlerts(
+  client: Client,
+  seed: string,
+  numHosts: number,
+  numDocs: number,
+  metadataIndex: string,
+  policyIndex: string,
+  eventIndex: string,
+  alertsPerHost: number,
+  options: TreeOptions = {}
+) {
+  const random = seedrandom(seed);
+  for (let i = 0; i < numHosts; i++) {
+    const generator = new EndpointDocGenerator(random);
+    await indexHostDocs(numDocs, client, metadataIndex, policyIndex, generator);
+    await indexAlerts(client, eventIndex, generator, alertsPerHost, options);
+  }
+  await client.indices.refresh({
+    index: eventIndex,
+  });
+  // TODO: Unclear why the documents are not showing up after the call to refresh.
+  // Waiting 5 seconds allows the indices to refresh automatically and
+  // the documents become available in API/integration tests.
+  await delay(5000);
```

Comment:
hmm this is odd. I do the same refresh you're doing here: https://github.com/elastic/kibana/blob/master/x-pack/test/api_integration/services/resolver.ts#L54

and I haven't seen any errors with those tests.

Are the alert tests failing in integration without the delay?

Reply:
Yeah they fail without the delay, the API calls return no results. Maybe the `Client`s are different somehow? I got errors when i tried to switch from `import { Client } from '@elastic/elasticsearch';` to `import { Client } from 'elasticsearch';` which is what es_archiver uses. I'm not sure what the differences are. And `client = getService('es')` might be doing yet another slightly different thing.